pypeit/core/pca.py

```diff
@@ -13,6 +13,34 @@
 
 from pypeit import debugger
 
+def func_vander(x, func, deg, minx=None, maxx=None):
+    if func == "polynomial":
+        return np.polynomial.polynomial.polyvander(x, deg)
+    elif func == "legendre":
+        if minx is None or maxx is None:
+            if np.size(x) == 1:
+                xmin, xmax = -1.0, 1.0
+            else:
+                xmin, xmax = np.min(x), np.max(x)
+        else:
+            xmin, xmax = minx, maxx
+        xv = 2.0 * (x-xmin)/(xmax-xmin) - 1.0
+        return np.polynomial.legendre.legvander(xv, deg)
+    elif func == "chebyshev":
+        if minx is None or maxx is None:
+            if np.size(x) == 1:
+                xmin, xmax = -1.0, 1.0
+            else:
+                xmin, xmax = np.min(x), np.max(x)
+        else:
+            xmin, xmax = minx, maxx
+        xv = 2.0 * (x-xmin)/(xmax-xmin) - 1.0
+        return np.polynomial.chebyshev.chebvander(xv, deg)
+    else:
+        msgs.error("Fitting function '{0:s}' is not implemented yet" + msgs.newline() +
+                   "Please choose from 'polynomial', 'legendre', 'chebyshev'")
+
+
```

Comment:
@profxj is this routine even being used?

Reply:
only if this out of date `pca` is called.

I suspect this module could be moved to `deprecated/`